oak_idl/src/lib.rs

```diff
@@ -125,97 +129,117 @@ impl Transport for Channel {
     }
 }
 
+/// A helper struct to facilitate building a [`Message`].
+///
+/// It delegates most methods to the underlying [`flatbuffers::FlatBufferBuilder`] instance, but it
+/// adds a [`MessageBuilder::finish`] method that returns a completed [`Message`] instance.
+///
+/// ```
+/// # struct Foo;
+/// #
+/// # impl flatbuffers::Verifiable for Foo {
+/// #     fn run_verifier(
+/// #         v: &mut flatbuffers::Verifier,
+/// #         pos: usize,
+/// #     ) -> Result<(), flatbuffers::InvalidFlatbuffer> {
+/// #         Ok(())
+/// #     }
+/// # }
+/// #
+/// # impl<'a> flatbuffers::Follow<'a> for Foo {
+/// #     type Inner = Self;
+/// #     fn follow(buf: &'a [u8], log: usize) -> Self::Inner {
+/// #         Self
+/// #     }
+/// # }
+/// #
+/// # impl Foo {
+/// #     pub fn create(b: &mut flatbuffers::FlatBufferBuilder) -> flatbuffers::WIPOffset<Foo> {
+/// #         flatbuffers::WIPOffset::new(0)
+/// #     }
+/// # }
+/// #
+/// let mut b = oak_idl::MessageBuilder::<Foo>::default();
+/// let v = b.create_vector::<u8>(&[14, 12]);
+/// let foo = Foo::create(&mut b);
```

Comment:
Is this a limitation of flatbuffers that the offset must be created separately? Is it not possible to have a trait that the type must implement so that it can be automatically called by `finish`.